tests/common_templates_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"os/exec"
+	"strings"
+	"net/http"
+	"io/ioutil"
+	"regexp"
+	"os"
+	"bufio"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Common templates", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	// Common templates read into string
+	var common_templates_yaml string
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+
+		// Setting namespace
+		_, err := exec.Command("/bin/bash", "-c", "/usr/bin/oc project " + tests.NamespaceTestDefault).Output()
+		tests.PanicOnError(err)
+
+		// Applying datavolume
+		// TODO1: This is currently bugged and replaced with using vm.yaml. 
+		// exec.Command("/bin/bash", "-c", "/usr/bin/oc apply -f tests/manifests/fedora_datavolume.yaml")
+
+		// Getting common_templates
+		// TODO2: replace downloading common-templates with getting common-templates from RPM
+		ct_yml_url := "https://github.com/kubevirt/common-templates/releases/download/v0.3.1/common-templates-v0.3.1.yaml"
+		response, err := http.Get(ct_yml_url)
+		tests.PanicOnError(err)
+		defer response.Body.Close()
+		data, err := ioutil.ReadAll(response.Body)
+		common_templates_yaml = string(data)
+
+	})
+
+	Context("Test loading Fedora", func() {
+
+		// CNV-1065
+		It("Test loading Fedora", func() {
+
+			// Patterns needed for parsing common-templates.yaml
+			source_pattern, _ := regexp.Compile("# Source: dist/templates/(.*\\.yaml)\\s*")
+			os_pattern, _ := regexp.Compile("([^-]+)-[^-]+-[^-]+\\.yaml")
+
+			for _, yaml_segment := range strings.Split(common_templates_yaml, "\n---\n") {
+
+				lines := strings.Split(yaml_segment, "\n")
+				// Skip this yaml segment if it doesn't contain VM template
+				if (len(lines) == 0 || ! source_pattern.MatchString(lines[0])) {
+					continue
+				}
+
+				filename := source_pattern.FindStringSubmatch(lines[0])[1]
+				os_name := os_pattern.FindStringSubmatch(filename)[1]
+				// Skip this template if it is not Fedora
+				if (os_name != "fedora") {
+					continue
+				}
+
+				By("Reading " + filename)
+				// Write data to a file and feed this file to oc process
```

Comment:
Martin said, that this test case could be some kind of concept. And in the next iteration it will be improved. But we (scale team) should study how we can work with testing infrastructure from our code in the future. For example, Martin prefers to test manifests from RPM files, but at the moment we don't have infrastructure for it.